pkg/provision/storage/commonStorage.go

```diff
@@ -71,8 +74,30 @@ func (p *CommonStorageProvisioner) ProvisionStorage(podAdditions *v1alpha1.PodAd
 	return nil
 }
 
-func (*CommonStorageProvisioner) CleanupWorkspaceStorage(workspace *dw.DevWorkspace, clusterAPI sync.ClusterAPI) error {
-	return runCommonPVCCleanupJob(workspace, clusterAPI)
+func (p *CommonStorageProvisioner) CleanupWorkspaceStorage(workspace *dw.DevWorkspace, clusterAPI sync.ClusterAPI) error {
+	_, totalWorkspaces, err := getSharedPVCWorkspaceCount(workspace.Namespace, clusterAPI)
+	if err != nil {
+		return err
+	}
+
+	// If the number of common + async workspaces that exist (started or stopped) is zero,
+	// delete common PVC instead of running cleanup job
+	if totalWorkspaces <= 1 {
+		existingPVC := &corev1.PersistentVolumeClaim{}
+		namespacedName := types.NamespacedName{Name: config.Workspace.PVCName, Namespace: workspace.Namespace}
+		err := clusterAPI.Client.Get(clusterAPI.Ctx, namespacedName, existingPVC)
+		if err != nil {
+			return err
+		}
+
+		err = clusterAPI.Client.Delete(clusterAPI.Ctx, existingPVC)
+		if err != nil && !k8sErrors.IsNotFound(err) {
+			return err
+		}
+		return nil
+	} else {
+		return runCommonPVCCleanupJob(workspace, clusterAPI)
```

Comment:
In terms of balancing conditional branches, it probably makes sense to move this one up to the top, i.e. have 
```go
if totalWorkspaces > 1 {
  return runCommonPVCCleanupJob(workspace, clusterAPI)
} else {
  // delete PVC
}
```